assembl/fabfile.py

```diff
@@ -1472,19 +1610,114 @@ def generate_dh_group():
+            else:
+                run("rm -f /etc/nginx/sites-enabled/%s" % (env.public_hostname))
+                run("ln -s /etc/nginx/sites-available/%(host_name)s /etc/nginx/sites-enabled/%(host_name)s" % {'host_name': env.public_hostname})
+                run("sudo /etc/init.d/nginx restart")
+    finally:
+        if debug:
+            os.unlink(file)
+
+
+def get_s3_file(bucket, key, destination=None):
+    import boto3
+    s3 = boto3.client('s3')
+    try:
+        ob = s3.get_object(Bucket=bucket, Key=key)
+        content = ob['Body'].read()
+        if destination:
+            with open(destination, 'w') as f:
+                f.write(content)
+        return content
+    except s3.exceptions.NoSuchKey:
+        return None
+
+
+@task
+def get_aws_localrc():
+    assert running_locally()
+    import requests
+    r = requests.get('http://169.254.169.254/latest/meta-data/iam/info')
+    assert r.ok
+    account = r.json()['InstanceProfileArn'].split(':')[4]
+    # This introduces a convention: local.rc files
+    # for a given amazon account will be stored in
+    # s3://bluenove-assembl-configurations/local_{account_id}.rc
+    content = get_s3_file(
+        'bluenove-assembl-configurations',
+        'local_%s.rc' % account,
+        env.projectpath + '/local.rc')
+    extends_re = re.compile(r'\b_extends\s*=\s*(\w+\.rc)')
+    match = extends_re.search(content)
+    while match:
+        key = match.group(1)
+        ex_content = get_s3_file('bluenove-assembl-configurations', key)
+        if not ex_content:
+            break
+        with open(key, 'w') as f:
+            f.write(ex_content)
+        match = extends_re.search(ex_content)
+    if not content:
+        raise RuntimeError("local_%s.rc was not defined in S3" % account)
+
+
+@task
+def setup_aws_default_region():
+    assert running_locally()
+    import requests
+    r = requests.get('http://169.254.169.254/latest/meta-data/placement/availability-zone')
+    assert r.ok
+    region = r.content[:-1]
+    run('aws configure set region ' + region)
+
+
+@task
+def aws_instance_startup():
+    """Operations to startup a fresh aws instance from an assembl AMI"""
+    if not env.projectpath:
+        # Assumption: Started from project directory
+        env.projectpath = os.cwd()
+    setup_aws_default_region()
+    get_aws_localrc()
+    if not exists(env.projectpath + "/local.rc"):
+        raise RuntimeError("Missing local.rc file")
+    env['rcfile'] = "local.rc"
+    load_rcfile_config()
+    aws_server_startup_from_local()
+
+
+@task
+def aws_server_startup_from_local():
+    """Update files that depend on local.rc and restart nginx, supervisor"""
+    create_local_ini()
+    venvcmd('assembl-ini-files populate %s' % (env.ini_file))
+    fill_template('assembl/templates/system/nginx_default.jinja2', env, 'var/share/assembl.nginx')
+    if is_supervisord_running():
+        supervisor_restart()
```

Comment:
Shouldn't you use `execute(supervisor_restart)` ?

Reply:
As I understood it, this was the first pass at migrating to cloud using fabric. The idea was to use fabric ON the machine itself. It was meant to only be run locally (on the machine). In any case, part of the dead code...